### PR TITLE
Implemented Basic CRUD Operations for User

### PR DESCRIPTION
**Pull request by hariprasath-r**

### CODE REVIEW

File: src/main/java/in/hp/java/userservice/service/UserService.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+    public void addUser(User user) {
+        userRepository.save(user);
+    }
+
+    public List<User> getUsers() {
+        return userRepository.findAll();
+    }
+
+    public User getUser(Long id) {
```
Comment (JayantPatil333) [inline src/main/java/in/hp/java/userservice/service/UserService.java]: What if we dont have that User id? please handle Exception using Controller Advise

File: src/main/java/in/hp/java/userservice/controller/UserController.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    @PutMapping
+    public ResponseEntity<Object> updateUser(@RequestBody User user) {
+        userService.updateUser(user);
+        return ResponseEntity.ok().build();
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Object> deleteUser(@PathVariable Long id) {
+        userService.deleteUser(id);
```
Comment (JayantPatil333) [inline src/main/java/in/hp/java/userservice/controller/UserController.java]: By this we are not able to send any customization message from Response Entity. This object should be generic so we can send different messages from different methods.

File: src/main/java/in/hp/java/userservice/model/User.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+import lombok.Data;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+@Entity
+@Data
+public class User {
```
Comment (JayantPatil333) [inline src/main/java/in/hp/java/userservice/model/User.java]: If this Model/DTO class then we should use it for Entity. Please create separate DTO and Entity.

File: src/main/java/in/hp/java/userservice/controller/UserController.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+import in.hp.java.userservice.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+public class UserController {
```
Comment (JayantPatil333) [inline src/main/java/in/hp/java/userservice/controller/UserController.java]: It if better practice to create interface for Controller which will contain configuration related APIs. And Class which will have only business logic. So in future if we want to add swagger configuration so we can add it in interface and our class is clean to see only business logic.

**Review by JayantPatil333 — COMMENTED** [review]
Please write test cases as well.

File: src/main/java/in/hp/java/userservice/service/UserService.java (hunk 1 above)
Comment (hariprasath-r) [reply src/main/java/in/hp/java/userservice/service/UserService.java]: Added Exception handlers

File: src/main/java/in/hp/java/userservice/controller/UserController.java (hunk 2 above)
Comment (hariprasath-r) [reply src/main/java/in/hp/java/userservice/controller/UserController.java]: Added generic response object

File: src/main/java/in/hp/java/userservice/model/User.java (hunk 3 above)
Comment (hariprasath-r) [reply src/main/java/in/hp/java/userservice/model/User.java]: Added UserDto

File: src/main/java/in/hp/java/userservice/controller/UserController.java (hunk 4 above)
Comment (hariprasath-r) [reply src/main/java/in/hp/java/userservice/controller/UserController.java]: Added interface implementation for UserController

**Pull request merged**